src/network/networkd-network.c

```diff
@@ -150,10 +169,21 @@ static int network_verify(Network *network) {
         if (network->ip_masquerade)
                 network->ip_forward |= ADDRESS_FAMILY_IPV4;
 
-        if (network->mtu > 0 && network->dhcp_use_mtu) {
-                log_warning("%s: MTUBytes= in [Link] section and UseMTU= in [DHCP] section are set. "
-                            "Disabling UseMTU=.", network->filename);
-                network->dhcp_use_mtu = false;
+        network_set_stacked_netdevs_mtu(network);
+
+        if (network->mtu > 0) {
+                if (network->dhcp_use_mtu) {
+                        log_warning("%s: MTUBytes= in [Link] section and UseMTU= in [DHCP] section are set. "
+                                    "Disabling UseMTU=.", network->filename);
+                        network->dhcp_use_mtu = false;
+                }
+
+                if (network->mtu < network->stacked_netdevs_mtu) {
+                        log_notice("%s: MTUBytes= must be equal or larger than vlan's MTUBytes= plus 4bytes or macvlan's MTUBytes=, "
+                                   "increasing MTUBytes= to %"PRIu32".",
+                                   network->filename, network->stacked_netdevs_mtu);
```

Comment:
What about having `network_set_stacked_netdevs_mtu` also return the stacked netdev identifier, and here saying something like:
`%s: bumping MTUBytes= from %"PRIu32" to %"PRIu32" because of stacked device %s", ...`. Then we don't need to go into the details of the calculation, and give more info to the user why the bump happened.